### PR TITLE
tests: Check that Icarus can parse arch sim models

### PR DESCRIPTION
**Pull request by daveshah1**
Should avoid problems like those fixed in #1137 in the future.

### CODE REVIEW

File: tests/arch/run-test.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+set -e
+
+echo "Running syntax check on arch sim models"
+iverilog -t null ../../techlibs/achronix/speedster22i/cells_sim.v
```
Comment (eddiehung) [inline tests/arch/run-test.sh]: Can we wildcard this? It would mean we don't need to remember to update this file when we add new arches

Comment (daveshah1) [reply tests/arch/run-test.sh]: Yes, I started this not sure how different arches specified these files, but then it seems like they are fairly consistent so just looking for cells_sim in root or subfolders should work

Comment (daveshah1) [reply tests/arch/run-test.sh]: Changed to use find and xargs

**Comment by eddiehung** [comment]
We might as well also chuck techlibs/common/sim{cells,lib}.v in here as well.

**Pull request merged**